pepper-apis/src/main/java/org/broadinstitute/ddp/util/QuestionUtil.java

```diff
@@ -23,13 +24,20 @@ public static boolean isReadonly(Handle handle, QuestionDto dto, String activity
             return false;
         }
 
-        InstanceStatusType statusType = handle.attach(JdbiActivityInstance.class)
-                .getByActivityInstanceGuid(activityInstanceGuid)
-                .map(ActivityInstanceDto::getStatusType)
-                .orElseThrow(() -> new DaoException(String.format(
+        ActivityInstanceDto activityInstanceDto = handle.attach(JdbiActivityInstance.class)
+                .getByActivityInstanceGuid(activityInstanceGuid).orElseThrow(() -> new DaoException(String.format(
                         "Could not find activity instance with guid=%s while getting question with id=%d and stableId=%s",
                         activityInstanceGuid, dto.getId(), dto.getStableId())));
 
-        return InstanceStatusType.COMPLETE.equals(statusType);
+        InstanceStatusType statusType = activityInstanceDto.getStatusType();
+        if (InstanceStatusType.COMPLETE.equals(statusType)) {
+            return true;
+        }
+
+        // If it is not the first instance - block editing
```

Comment:
I see. Typically, if question is write-once, then we will have already marked it read-only when status is completed. However, since there can be multiple instance of activity, the second/third/etc instances becomes a "loophole" where user can edit the question answer. So this will prevent that.

I think there is an interesting case where question is configured write-once BUT answers are not copied from previous instance. So in that case, user somehow somewhere gets a new blank instance, and then for that particular question they won't be able to answer it since it's now read-only. We can probably improve on this and fix that edge-case down the line.